### PR TITLE
tfstate: add dependencies

### PR DESCRIPTION
**Pull request by tormath1**
dependencies will be based on the interpolation map(https://github.com/cycloidio/terracognita/pull/81) created during the import

### CODE REVIEW

**Comment by tormath1** [comment]
@xescugc @talset ready for reviewing :)

File: state/writer.go [hunk 1]
```diff
@@ -112,5 +114,88 @@ func (w *Writer) Sync() error {
...
+			}
+		}
+	}
+
+}
+
+// extractResourceTypeAndName will parse a TF variable to return
+// the resource type and the name of the resource
+func extractResourceTypeAndName(value string) (string, string) {
+	res := regexp.MustCompile(`\${(.+)\.(.+)\.(.+)}`)
```
Comment (kerak19) [inline state/writer.go:185]: You could compile this regex only once and reuse

File: state/writer.go [hunk 2]
```diff
@@ -16,6 +16,8 @@ import (
 	"github.com/pkg/errors"
 )
 
+var res = regexp.MustCompile(`\${(.+)\.(.+)\.(.+)}`)
```
Comment (kerak19) [inline state/writer.go:19]: Maybe it's also worth adding some docs/better name to it, so it's known what does it match.

Comment (tormath1) [reply state/writer.go]: @kerak19 done

**Comment by kerak19** [comment]
RS

**Comment by tormath1** [comment]
I RS with #133 in order to allow the activation / deactivation of this feature. A test case has been added too.

**Review by xescugc — APPROVED** [review]
LGTM

**Pull request merged**